src/mono/sample/wasm/browser-logprofile/LogProfiler.cs

```diff
@@ -0,0 +1,20 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Mono.Profiler.Log
+{
+	/// <summary>
+	/// Internal calls to match with https://github.com/dotnet/runtime/blob/release/6.0/src/mono/mono/profiler/log.c#L4061-L4097
+	/// </summary>
+	internal class LogProfiler
+	{
+		[MethodImplAttribute(MethodImplOptions.InternalCall)]
```

Comment:
Let's get rid of this icall and native method.
See AOT profiler, it would stop when it sees registered C# method called. It could also flush at that point.

Reply:
Exposing the log profiler to `aot.c` will require exposing/duplicating https://github.com/dotnet/runtime/blob/6092b21d6e74a343f6019551c64723d7aced7c2e/src/mono/mono/profiler/log.c#L286 in the same way it's done for the AOT profiler, but the structure a significantly more complex... 

Should we expose a single "takeHeapshot" static mono method that aot.c can use?

Reply:
sounds good to me

cc @lambdageek

Reply:
This was changed to have jit interception done directly in the log profiler, instead of using what's in aot.c.